doc/changelog.md

```diff
@@ -137,75 +227,81 @@
+  would no longer decompose the `PSWAP` and `ISWAP` gates.
 
-* Add support for the `BlockEncode` operator in Catalyst.
+* The `qml.BlockEncode` operator is now supported with Catalyst.
   [(#483)](https://github.com/PennyLaneAI/catalyst/pull/483)
 
-* Remove copies of TOML device configuration files for Lightning device in Catalyst.
-  [(#472)](https://github.com/PennyLaneAI/catalyst/pull/472)
+* Additional debugging functions are now available in the `catalyst.debug` directory.
+  [(#529)](https://github.com/PennyLaneAI/catalyst/pull/529)
+  [(#522)](https://github.com/PennyLaneAI/catalyst/pull/531)
 
-* Remove `qextract_p` and `qinst_p` from forced-order primitives.
-  [(#469)](https://github.com/PennyLaneAI/catalyst/pull/469)
+  This includes:
 
-* Update `AllocateQubit` and `AllocateQubits` in `LightningKokkosSimulator` to preserve
-  the current state-vector before qubit re-allocations.
-  [(#479)](https://github.com/PennyLaneAI/catalyst/pull/479)
+  - `compile_from_mlir(ir[, compiler, ...])` to compile a Catalyst function to binary code from the
+    provided MLIR.
 
-* Add support for post-selection in mid-circuit measurements. This is currently supported only on Lightning simulators.
-  [(#491)](https://github.com/PennyLaneAI/catalyst/pull/491)
+  - `filter_static_args(args, static_argnums)` to remove static values from arguments using the
+    provided index list.
 
-  This is an example of post-selection usage:
+  - `get_cmain(fn, *args)` to return a C program that calls a jitted function with the provided
+    arguments.
 
-  ```py
-  import pennylane as qml
-  from catalyst import qjit
+  - `print_compilation_stage(fn, stage)` to print one of the recorded compilation stages for a
+    JIT-compiled function.
 
-  dev = qml.device("lightning.qubit", wires=1)
+  - `promote_arguments(target_signature, args)` to promote arguments to the provided target
+    signature, preserving PyTrees.
 
-  @qjit
-  @qml.qnode(dev)
-  def f():
-      qml.Hadamard(0)
-      m = measure(0, postselect=1)
-      return qml.expval(qml.PauliZ(0))
-  ```
+  For more details, please see the `catalyst.debug` documentation.
 
-* Add support for reset option in mid-circuit measurements.
-  [(#507)](https://github.com/PennyLaneAI/catalyst/pull/507)
+* Remove copies of TOML device configuration files for Lightning device in Catalyst.
+  [(#472)](https://github.com/PennyLaneAI/catalyst/pull/472)
 
-  This is an example of reset usage:
+* Capturing quantum circuits with many gates prior to compilation is now quadratically faster (up to
+  a factor), by removing `qextract_p` and `qinst_p` from forced-order primitives.
+  [(#469)](https://github.com/PennyLaneAI/catalyst/pull/469)
 
-  ```py
-  dev = qml.device("lightning.qubit", wires=1)
+* Update `AllocateQubit` and `AllocateQubits` in `LightningKokkosSimulator` to preserve
+  the current state-vector before qubit re-allocations.
+  [(#479)](https://github.com/PennyLaneAI/catalyst/pull/479)
```

Comment:
Ooops! I didn't get any GH notification for this comment 2d ago! :thinking: 

This PR is more like a technical update to the LightningKokkos support. It doesn't have any user impact as this is related to the dynamic qubits allocations/deallocations tickets. Secondary calls to `AllocateQubit` and `AllocateQubits` in LightningKokkosSimulator used to fully erase the previously allocated state-vector, but now second calls to these methods will update the previous state-vector to preserve the state at anytime within the scope of programs. This is not supported from the frontend and compiler at the moment and doesn't have any user impact. It can be also removed from the changelog, I guess.

Reply:
```suggestion
* Update `AllocateQubit` and `AllocateQubits` in `LightningKokkosSimulator` to preserve
  the current state-vector before qubit re-allocations in the runtime dynamic qubits management.
  [(#479)](https://github.com/PennyLaneAI/catalyst/pull/479)
```

Reply:
Using dynamic qubit allocation (not accessible to PennyLane users) used to erase the state of pre-existing qubits. This is no longer the case.